docs/fsharp/style-guide/formatting.md

```diff
@@ -100,42 +100,52 @@ If you have a long function definition, place the parameters on new lines and in
 
 ```fsharp
 module M =
-    let LongFunctionWithLotsOfParameters(aVeryLongParam: AVeryLongTypeThatYouNeedToUse)
-                                        (aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse)
-                                        (aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse)
-                                        =
-        // ... the body of the method follows
+    let LongFunctionWithLotsOfParameters
+        (aVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        (aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        (aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse)
+        =
+            // ... the body of the method follows
 ```
 
 This also applies to members, constructors, and parameters using tuples:
 
 ```fsharp
 type TM() =
-    member _.LongMethodWithLotsOfParameters(aVeryLongParam: AVeryLongTypeThatYouNeedToUse,
-                                            aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse,
-                                            aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse) =
-        // ... the body of the method
-
-type TC(aVeryLongCtorParam: AVeryLongTypeThatYouNeedToUse,
+    member _.LongMethodWithLotsOfParameters
+        (
+            aVeryLongParam: AVeryLongTypeThatYouNeedToUse,
+            aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse,
+            aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse
+        ) =
+            // ... the body of the method
```

Comment:
IMHO this looks less noisy, and seems to be more consistent with the formatting of curried parameters as well as with how Fantomas currently formats similar stuff:

```suggestion
    member _.LongMethodWithLotsOfParameters
        (aVeryLongParam: AVeryLongTypeThatYouNeedToUse,
         aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse,
         aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse) =
            // ... the body of the method
```

Also note the placement of `=` on the last parameter line, in line with the following part of the docs:

> If the parameters are currified or there's an explicit return type annotation, it is preferable to place the `=` character on a new line

Reply:
The downside to the approach you mention @cmeeren is that it doesn't distinguish the body of the member from the parameter list very well.

Reply:
I agree, and personally I always intent the parameter list one level extra (one level more than the body). How about that?

```suggestion
    member _.LongMethodWithLotsOfParameters
            (aVeryLongParam: AVeryLongTypeThatYouNeedToUse,
             aSecondVeryLongParam: AVeryLongTypeThatYouNeedToUse,
             aThirdVeryLongParam: AVeryLongTypeThatYouNeedToUse) =
        // ... the body of the method
```

Reply:
I think double indentation would be a very special, hard to follow manually, rule. Another downside of your approach @cmeeren is that it has more chances of generating bigger diffs when adding parameters.

Reply:
> I think double indentation would be a very special, hard to follow manually, rule.

Hard to remember, you mean? I have to admit that I'm coming at this from the angle of making Fantomas avoid vanity alignment and thus have a tooling mindset, but IMHO it's still much easier to do and maintain manually than the current vanity alignment recommendation, so I'm not sure I buy the argument.

> Another downside of your approach @cmeeren is that it has more chances of generating bigger diffs when adding parameters.

Are you talking about the fact that the parameter lines are not "equal" in that the last one has a parenthesis? If so, then 1) I don't see how that creates big diffs, and 2) certainly not bigger than existing similar recommendations like records:

```f#
type MyRecord =
    { A: int;
      B: string;
      C: byte [] }
```

---

In any case, I can live with your suggestion. But even that may have the problem @cartermp mentioned above. I'll make a comment on the relevant code to show what I mean.

Reply:
wrt diffs, I know it's not very common but you could add parameters to the front, and with the proposed approach there would be only n number of diff lines for n new parameters, while you would get n+1 with your approach.

> Hard to remember, you mean?

Yeah my aim is to always use fantomas as well, but I don't want fantomas to dictate a format that is hard to remember or apply manually, otherwise you will always depend on it. I mean, ideally fantomas should dictate a format that is easy to remember so that you can progressively learn it and come out naturally at some point (and your CI builds start to be green from the start, when fantomas is run in your PRs or feature branches).

Reply:
yeah, i'd prefer no double indentation here.

Reply:
Let's keep the current PR suggestion, then. 🙂 Though note that `// ... the body of the method` is indented incorrectly and should be fixed.

Reply:
Thing is, my opinion about the double-indentation on the parameters is not the same as the extra indentation for the body. The latter feels visually better because it adds one extra level after the parameters (so, parameters -> +1, body -> +2).

Reply:
(while you were suggesting `parameters -> +2; body -> +1` before)

Reply:
Are you saying that if the parameters are indented, the body should be doubly indented? If you intend that to be part of the PR, that should be made explicit.

Personally I think the body should be kept at one level of indentation. The `=` on a separate line (with or without type annotation) creates sufficient separation. @cartermp any opinions?

Reply:
@knocte @cartermp The four last comments above relate to the double body indentation.

I currently stand by my view that double-indenting the body is unnecessary (and AFAIK without precedent elsewhere in the style guide), and that there is already sufficient separation between the parameters and the body.

@knocte If you agree, or if @cartermp agrees and decides on keeping bodies single-indented, it should be simple to just go through my suggestions here in this PR and commit them.

Alternatively, if you both agree that bodies should in fact be doubly indented in these cases, ignore my change suggestions, but please make it explicit somewhere that this should be done and why.